Tests/IntegrationTests/Roslyn/RoslynGenTests.cs

```diff
@@ -0,0 +1,191 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="RoslynGenTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NuGet;
+using SonarQube.Plugins.Roslyn;
+using SonarQube.Plugins.Test.Common;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace SonarQube.Plugins.IntegrationTests
+{
+    [TestClass]
+    public class RoslynGenTests
+    {
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        public void RoslynPlugin_GenerateForValidAnalyzer_Succeeds()
```

Comment:
Finally, the end to end test. It doesn't execute the plugin generator exe directly as that would mean hitting the public NuGet site to download a real analyzer directly. This seems slightly unfair to the package author as it would give a very misleading idea of the number of real users of their analyzer.